File: cliente_js/cliente.js
const net = require('net');
const readline = require('readline');

const client = new net.Socket();
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const PORT = 8080; // Asegúrate de que este puerto coincida con el del servidor.
const HOST = '127.0.0.1'; // Asumiendo que el servidor está en la misma máquina.

client.connect(PORT, HOST, () => {
    console.log(`Conectado al servidor en ${HOST}:${PORT}`);
});

client.on('data', (data) => {
    const response = data.toString(); // Convertir el buffer en string
    console.log(`Servidor: ${response}`);

    if (response.includes('Seleccione una opción')) {
            if (opcion === '1' || opcion === '2') {
                client.write(`${opcion}\n`); // Enviar opción al servidor
            } else {
                console.log('Opción no válida.');
                client.write('Invalid\n'); // Informar al servidor que la opción es inválida
            }
    } else if (response.includes('Ingrese la longitud')) {
        rl.question('Ingresa la longitud deseada: ', (longitud) => {
            const mensaje = `${longitud}\n`; // Asegúrate de terminar con un salto de línea
            client.write(mensaje);
        });
    } else {
        // Mostrar la respuesta final y cerrar la conexión
        console.log(`Resultado final: ${response}`);
        rl.close();
        client.end(); // Cerrar la conexión del cliente
    }
});

client.on('close', () => {
    console.log('Conexión cerrada.');
});

client.on('error', (err) => {
    console.error(`Error de conexión: ${err.message}`);
});

File: cliente_js/cliente.py
import socket

def main():
    HOST = '127.0.0.1'  # Dirección IP del servidor
    PORT = 8080          # Puerto del servidor

    # Crear un socket TCP/IP
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.connect((HOST, PORT))
        print(f'Conectado al servidor en {HOST}:{PORT}')
        
        while True:
            try:
                # Recibir datos del servidor
                data = s.recv(1024)
                if not data:
                    break

                # Intentar decodificar los datos recibidos
                try:
                    response = data.decode('utf-8')
                except UnicodeDecodeError:
                    response = data.decode('latin-1')  # Intentar con una codificación diferente

                print(f'Servidor: {response}')

                if 'Seleccione una opción' in response:
                    opcion = input('Elige una opción: (1) Generar nombre de usuario (2) Generar contraseña: ')
                    if opcion == '1' or opcion == '2':
                        s.sendall((opcion + '\n').encode('utf-8'))
                    else:
                        print('Opción no válida.')
                        continue

                elif 'Ingrese la longitud' in response:
                    longitud = input('Ingresa la longitud deseada: ')
                    s.sendall((longitud + '\n').encode('utf-8'))

                else:
                    # Mostrar la respuesta final y salir
                    print(f'Resultado final: {response}')
                    break

            except Exception as e:
                print(f'Error: {e}')
                break

if __name__ == "__main__":
    main()
